fake_spectra/spectra.py

```diff
@@ -619,18 +635,32 @@ def replace_not_DLA(self, ndla, thresh=10**20.3, elem="H", ion=1):
         found += np.size(ind)
         self.discarded = self.NumLos-np.size(ind)
         print("Discarded: ",self.discarded)
+        #dom = 0
         while found < wanted:
+        #while dom<1000:
+            size_ind = None
             #Get a bunch of new spectra
             self.cofm = self.get_cofm()
             col_den = self.compute_spectra(elem,ion,1215,False)
-            ind = self.filter_DLA(col_den, thresh)
+            cdsum = np.sum(col_den, axis = 1)
+            col_den_added = np.zeros_like(cdsum)
+            #print('\n Col Density is :{sd}'.format(sd=np.sum(col_den, axis=1)))
+            #ind = self.filter_DLA(col_den, thresh)
+            
+            ### Call manager rank here
+            self.comm.Reduce(cdsum, col_den_added, op=MPI.SUM, root=0)
```

Comment:
You could put this and the above behind a condition:
if self.MPI:
(self.comm.Reduce...)